internal/cloudapi/v2/middleware.go

```diff
@@ -29,6 +32,32 @@ func (s *Server) TenantChannelMiddleware(next echo.HandlerFunc) echo.HandlerFunc
 	}
 }
 
+type ComposeHandlerFunc func(ctx echo.Context, id string) error
+
+// Ensures that the job's channel matches the JWT cannel set in the echo.Context
+func (s *Server) EnsureJobChannel(next ComposeHandlerFunc) ComposeHandlerFunc {
+	return func(c echo.Context, id string) error {
+		jobId, err := uuid.Parse(id)
+		if err != nil {
+			return HTTPError(ErrorInvalidComposeId)
+		}
+
+		jobChannel, err := s.workers.JobChannel(jobId)
+		if err != nil {
+			fmt.Println(err)
+			return HTTPErrorWithInternal(ErrorComposeNotFound, err)
+		}
+
+		ctxChannel := c.Get(TenantChannelKey).(string)
+		if jobChannel != ctxChannel {
+			return HTTPErrorWithInternal(ErrorResourceNotFound,
```

Comment:
This should be also `ErrorResourceNotFound`, I think we should make sure not to leak that the job exists if there is a tenant mismatch. (So let's even maybe return the exact same `HTTPErrorWithInternal` call?)

Reply:
> This should be also `ErrorResourceNotFound`, I think we should make sure not to leak that the job exists if there is a tenant mismatch.

Did you mean `ErrorComposeNotFound`, instead of `ErrorResourceNotFound`?

Nevertheless I agree and my intention was explicitly to not leak any information, but I apparently misunderstood how  `HTTPErrorWithInternal` works. After reading more about it, I think that returning `HTTPError(ErrorComposeNotFound)` should be more appropriate.

> (So let's even maybe return the exact same `HTTPErrorWithInternal` call?)

I'm not sure what do you mean by this suggestion. `HTTPErrorWithInternal` expects a second argument and if we do not provide additional information, there is nothing to pass 🤔

Reply:
I reworked the code to return `HTTPError(ErrorComposeNotFound)` in the latest push.

Reply:
`HTTPError(ErrorComposeNotFound)` is indeed the best option imho, thanks! :)